freeipa/src/main/java/com/sequenceiq/freeipa/service/user/UsersyncService.java

```diff
@@ -0,0 +1,93 @@
+package com.sequenceiq.freeipa.service.user;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.client.HttpClientConfig;
+import com.sequenceiq.cloudbreak.orchestrator.model.GatewayConfig;
+import com.sequenceiq.freeipa.api.model.users.Group;
+import com.sequenceiq.freeipa.api.model.users.SynchronizeUsersRequest;
+import com.sequenceiq.freeipa.api.model.users.User;
+import com.sequenceiq.freeipa.client.FreeIpaClient;
+import com.sequenceiq.freeipa.client.FreeIpaClientBuilder;
+import com.sequenceiq.freeipa.client.model.RPCResponse;
+import com.sequenceiq.freeipa.entity.FreeIpa;
+import com.sequenceiq.freeipa.entity.Stack;
+import com.sequenceiq.freeipa.service.FreeIpaService;
+import com.sequenceiq.freeipa.service.GatewayConfigService;
+import com.sequenceiq.freeipa.service.TlsSecurityService;
+import com.sequenceiq.freeipa.service.stack.StackService;
+
+@Service
+public class UsersyncService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UsersyncService.class);
+
+    private static final String STACK_NOT_FOUND_ERROR = "Stack not found for environment %s and name %s";
+
+    private static final String FREEIPA_NOT_FOUND_ERROR = "FreeIPA not found for stack %d";
+
+    @Inject
+    private GatewayConfigService gatewayConfigService;
+
+    @Inject
+    private StackService stackService;
+
+    @Inject
+    private FreeIpaService freeIpaService;
+
+    @Inject
+    private TlsSecurityService tlsSecurityService;
+
+    public void synchronizeUsers(SynchronizeUsersRequest request) {
+        LOGGER.info("UsersyncService.synchronizeUsers() called");
+
+        Stack stack = stackService.getByNameAndEnvironment(request.getName(), request.getEnvironmentName());
```

Comment:
freeipa client creation maybe should be moved to a separate class, so we don't have to copy these 5 lines everywhere.